crates/sui-core/src/authority_server.rs

```diff
@@ -372,39 +364,24 @@ impl ValidatorService {
             metrics,
             traffic_controller: _,
             client_id_source: _,
-            validator_tx_finalizer,
         } = self.clone();
         let transaction = request.into_inner();
         let epoch_store = state.load_epoch_store_one_call_per_task();
 
         transaction.validity_check(epoch_store.protocol_config(), epoch_store.epoch())?;
 
-        // When authority is overloaded and decide to reject this tx, we still lock the object
-        // and ask the client to retry in the future. This is because without locking, the
-        // input objects can be locked by a different tx in the future, however, the input objects
-        // may already be locked by this tx in other validators. This can cause non of the txes
-        // to have enough quorum to form a certificate, causing the objects to be locked for
-        // the entire epoch. By doing locking but pushback, retrying transaction will have
-        // higher chance to succeed.
-        let mut validator_pushback_error = None;
-        let overload_check_res = state.check_system_overload(
-            &consensus_adapter,
-            transaction.data(),
-            state.check_system_overload_at_signing(),
-        );
-        if let Err(error) = overload_check_res {
-            metrics
-                .num_rejected_tx_during_overload
-                .with_label_values(&[error.as_ref()])
-                .inc();
-            // TODO: consider change the behavior for other types of overload errors.
-            match error {
-                SuiError::ValidatorOverloadedRetryAfter { .. } => {
-                    validator_pushback_error = Some(error)
```

Comment:
We don't need this anymore because validator tx finalizer on the validators that signed this will drive it to finality eventually anyway. So it's fine to early return on error.

Reply:
I'm not convinced that our delays are small enough to make this code unnecessary.

Reply:
Removed